mypy/test/helpers.py

```diff
@@ -312,6 +255,55 @@ def retry_on_error(func: Callable[[], Any], max_wait: float = 1.0) -> None:
             time.sleep(wait_time)
 
 
+class AssertionFailure(Exception):
```

Comment:
You could probably use the built-in exception `AssertionError` instead of defining a new class here.